### ISSUE TITLE
[Bug] (timed transition back to Main) Uncaught TypeError: this.moduleTransition is not a function

### ISSUE THREAD

**Issue by berlincount** [shbatm/MMM-Carousel#38]
**Describe the bug**

Chromium says:

```
MMM-Carousel.js:481 Uncaught TypeError: this.moduleTransition is not a function
    at Class.transitionTimeoutCallback (MMM-Carousel.js:481:10)
    at MMM-Carousel.js:263:54
````

**To Reproduce**

on commit 22b64b2aa686b6d3c1972aef770be64082dd5f89

````
    {
      module: 'MMM-Carousel',
      position: 'bottom_bar', // Required only for navigation controls
      config: {
        transitionInterval: 0,
        transitionTimeout: 60000,
        homeSlide: "Main",
        showPageIndicators: true,
        showPageControls: true,
        ignoreModules: ['clock', 'alert', 'calendar'],
        mode: 'slides',
        slides: {
          "Main": [...]
          "Calendar": [...],
        }
      }
    },
````

... switch to Calendar slide ...

**Expected behavior**

... never returns to Main slide ...

**Comment by berlincount** [comment]
this is still a problem with commit c120cb68c26f33c5d4be8d2f9ceadce4ac3e8fb7 :-/

what am I doing wrong?

**Comment by KristjanESPERANTO** [comment]
I can't see any issue with your config. But please try this simple one if the issue occurs with that too:

```js
    {
      module: "MMM-Carousel",
      position: "bottom_bar",
      config: {
        transitionInterval: 5000,
        transitionTimeout: 0,
        showPageIndicators: true,
        showPageControls: true,
        ignoreModules: ["alert"],
        mode: "slides",
        slides: {
          "Clock": ["clock"],
          "Calendar": ["calendar"]
        }
      }
    },
```

If the issue still occurs: Update the module, show us which version of MagicMirror, node (`node -v`) and npm (`npm -v`) you use.

> Commit this is still a problem with commit https://github.com/shbatm/MMM-Carousel/commit/c120cb68c26f33c5d4be8d2f9ceadce4ac3e8fb7 :-/

This commit should not have caused any impact on the function of the module. It's just cosmetic changes.

**Comment by KristjanESPERANTO** [comment]
Oh, I didn't read your text carefully enough. It's about the main slide. Why did you set `transitionInterval` to 0?

**Comment by berlincount** [comment]
> Oh, I didn't read your text carefully enough. Is it your intention that you can only switch slides manually?

yes, I only want to change the slide on click / touch ...

**Comment by KristjanESPERANTO** [comment]
I see the issue now, but haven't got a fix yet. I also have it with the status of 2021 (before all my changes), the module already behaved like this. I don't know if I'll find time to work on it in the near future.

**Comment by berlincount** [comment]
I dug around a bit, and found that both in the `setTimout()` call in lines 267 and 465 _and_ in the definition of `transitionTimeoutCallback` the arrow operator (`() => {...}`) is being used. Not sure it should be like that, but my trying around only changed the error messages so far ...

**Comment by berlincount** [comment]
okay, after digging around for a while - and liberally throwing around `console.log()` statements - I'm thorougly confused, and thing it's a multi-layered scoping issue beyond my ability to solve easily.

I'm not sure whether there's any guide or graph showing the contexts of various objects in MMM, but ... getting lost seems very easy 🤪

**Comment by KristjanESPERANTO** [comment]
@berlincount I think @btastic fixed it with #66 😃. Please check with the new version.

**Comment by berlincount** [comment]
oh wow, indeed - interesting that I couldn't find that fix. Thank you, @btastic!
